controllers/buildpipeline/buildpipeline_adapter.go

```diff
@@ -108,6 +108,36 @@ func (a *Adapter) EnsureSnapshotExists() (result controller.OperationResult, err
 		return controller.ContinueProcessing()
 	}
 
+	if _, found := a.pipelineRun.ObjectMeta.Annotations[tekton.SnapshotNameLabel]; found {
+		a.logger.Info("The build pipelineRun is already associated with existing Snapshot via annotation",
+			"snapshot.Name", a.pipelineRun.ObjectMeta.Annotations[tekton.SnapshotNameLabel])
+		canRemoveFinalizer = true
+		return controller.ContinueProcessing()
+	} else {
```

Comment:
We can make it more readble by not using `else` as `if` is returning from function

```
if annotated {
  return
}

existingSnaphsots ...
```

Reply:
Agreed, updated.